src/main/java/com/lineate/mdzinarishvili/reminderapp/services/ReminderService.java

```diff
@@ -1,4 +1,58 @@
 package com.lineate.mdzinarishvili.reminderapp.services;
 
+import com.lineate.mdzinarishvili.reminderapp.dao.ReminderDAO;
+import com.lineate.mdzinarishvili.reminderapp.exceptions.NotFoundException;
+import com.lineate.mdzinarishvili.reminderapp.models.Reminder;
+import org.springframework.stereotype.Service;
+
+import java.util.List;
+import java.util.Optional;
+
+@Service
 public class ReminderService {
+    private final ReminderDAO reminderDao;
+
+    public ReminderService(ReminderDAO reminderDao) {
+        this.reminderDao = reminderDao;
+    }
+
+    public List<Reminder> getReminders() {
+        return reminderDao.selectReminders();
+    }
+

```

Comment:
save and update methods should return saved ore updated dto with id from db

Reply:
I had a question about dtos. I usually use them when getting data from the user (with the @RequestBody annotation). Should I be using them to send and receive data? Should the dto also have an id attribute? Or were you referring to the fact that I should define something like an ApiResponse class and return that?

Reply:
The best practice here is:
Use DTOs (requests and responses) for the controller layer
Use models for the service layer
User entity for the repository layer
And map one to another using mapper.

Such as you are not using ORM and repository you can do something like:
Use the same DTOs (Reminder) in the service and DAO layer 
Use (ReminderRequest / ReminderResponse) DTOs in the controller layer

It means: 
Create ReminderRequest and use it as a request parameter for the controller endpoints.
pass it to the service layer and then map it to the Reminder model
after some work service layer returns that or a new Reminder from service
map that Reminder in the controller to ReminderResponse and return from the endpoint